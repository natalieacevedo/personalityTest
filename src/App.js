import * as React from "react";
import "./styles/main.scss";
import Home from "./components/Home";
import Questions from "./components/Questions";
import mockQuestions from "./mockData";
import Results from "./components/Results";
const { useState } = React;

function App() {
  //state to keep questions and answers got from 'backend'
  const [questions, setQuestions] = useState([]);
  const [results, setResults] = useState([]);
  const [finish, setFinish] = useState(false);

  function keepResults(ind, questionNumber) {
    setResults((previous) => {
      if (previous.length >= questionNumber) {
        let newArr = [...previous];
        newArr[questionNumber] = ind;
        console.log(newArr);
        return newArr;
      }
      return previous.concat([ind]);
    });
  }

  function populateQuestions() {
    //here we will make a 'GET API request'.
    setQuestions(mockQuestions);
  }

  function isFinished() {
    setFinish(!finish);
  }

  function goBackHome(e) {
    e.preventDefault();
    setQuestions([]);
    setResults([]);

    isFinished();
  }

  if (questions.length === 0) {
    return <Home populateQuestions={populateQuestions} />;
  } else {
    if (finish) {
      return <Results results={results} goBackHome={goBackHome} />;
    } else {
      return (
        <Questions
          questions={questions}
          keepResults={keepResults}
          results={results}
          isFinished={isFinished}
        />
      );
    }
  }
}

export default App;
