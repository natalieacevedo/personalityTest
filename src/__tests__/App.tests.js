import React from "react";
import App from "../App";
import { create } from "react-test-renderer";

describe("snapshot app component", () => {
  test("testing app component", () => {
    let tree = create(<App />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
