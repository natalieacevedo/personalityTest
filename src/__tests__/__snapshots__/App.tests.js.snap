// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot app component testing app component 1`] = `
<main
  className="mainPage"
>
  <h1
    className="main-title"
  >
    Test: Are you and introvert or an extrovert
  </h1>
  <p>
    So do you consider yourself more of an introvert or an extrovert? Take this test, put together by 
    <cite>
      psychoanalyst Natalie Acevedo
    </cite>
  </p>
  <img
    alt="headPuzzle"
    className="main-img"
    src="puzzle-head.png"
  />
  <button
    className="mainButton"
    onClick={[Function]}
  >
    Find out !
  </button>
</main>
`;
